frontend/src/hooks/useWebSocket.ts

```diff
@@ -32,7 +33,7 @@ interface SocketCallbacks<In, Out> {
  * `Out` - The type of data sent out over the websocket.
  */
 const useWebSocket = <In, Out>(
-  addr: string,
+  address: string | undefined,
```

Comment:
I just tweaked the logic so that if you pass `undefined` as the address, it doesn't connect.